internal/provider/aws/provider.go

```diff
@@ -509,7 +529,7 @@ func (p *awsProvider) ReconcileFirewallRules(
 			Groups:             groups,
 		})
 		if err != nil {
-			return status, fmt.Errorf("failed to modify network interface attribute: %w", err)
+			return status, conditions, fmt.Errorf("failed to modify network interface attribute: %w", err)
```

Comment:
Don't you want to set the condition v1alpha1.FirewallRuleConditionTypeNetworkInterfaceAssociated to Unknown here because of the error here ?